### PR TITLE
Identify Mode not persisted when map saved (#5566)

### PR DESCRIPTION
**Pull request by AGMETEOR**
## Description
This PR ensures that when a user selects identify mode from BurgerMenu > Settings as either Click or Hover, that mode is persisted with the data resource data.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Enhancement to existing feature

# Issue

**What is the current behavior?**

#5566 

**What is the new behavior?**
Identify tool mode is persisted. For example if a user selects Hover, it's persisted with that particular map resource

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
How to test thing feature
- Import a map with layers in MapStore
- Click Burger menu > Settings
- Click on `Trigger event for the display of the information sheet` and select either Hover or Click and close that modal
- Click on BurgerMenu > Save
- If you navigate away, the hover mode (if was selected) should still be activated

### CODE REVIEW

**Comment by AGMETEOR** [comment]
> Good catch about the point used.
> Anyway mouseMoveListener is used essentially for performance reasons (having continuous mouse move action when not catched was causing a lot of unnecessary load).
> Moreover there are some other tools that can register to listen mouse move event, so you may have to make you sure that `identifyFloatingTool` is registered.
> 
> Anyway I see this is the only flag used in the current state to identify when the event is registered, and this is probably a little unclear or unoconsistent.
> For this reason I'd like to ask you to modify a little your PR in order to improve and make more clear the whole system.
> 
> There is an entry in the map saved called `mapInfoConfiguration` (and in the state inside `mapInfo.configuration`). I think you should save an a new entry here to save this information, it looks the correct place to me. (e,g, `trigger: 'hover'|'click'`, if missing click is understood). Saving the event listener is a little unclear and may also be difficoult to maintain in the future.
> 
> The action that changes the configuration, may also trigger the event listener register/unregister as a side effect (now is done by `onTriggerChange`) instead of being controlled by this (unclear) variable.
> On saved map load, this configuration must be read to setup the event listener correctly as well
> 
> Please during your develoment also check if:
> 
> * the floating identify tool works also in geostore after these changes.
> * switching between maps

Hello @offtherailz , I have changed the implementation according to this review.

File: web/client/epics/map.js [hunk 1]
```diff
@@ -93,6 +95,15 @@ const handleCreationLayerError = (action$, store) =>
             ]) : Rx.Observable.empty();
         });
 
+// Epic to determine which trigger to use when map loads
```
Comment (offtherailz) [inline web/client/epics/map.js:98]: This epic have to be moved go in mapInfo epics. 
I suggest to intercept **also** on the `setMapTrigger` event, so we have a single entry point  to register or unregister the identify tool, and you can remove the call the registerEventListener from the component.

File: web/client/plugins/Identify.jsx [hunk 2]
```diff
@@ -233,6 +233,7 @@ const FeatureInfoFormatSelector = connect((state) => ({
 const FeatureInfoTriggerSelector = connect((state) => ({
     trigger: isMouseMoveIdentifyActiveSelector(state) ? 'hover' : 'click'
 }), {
+    onSetMapTrigger: setMapTrigger,
     onTriggerChange: (event) => event.target.value === 'hover' ? registerEventListener('mousemove', 'identifyFloatingTool') : unRegisterEventListener('mousemove', 'identifyFloatingTool')
```
Comment (offtherailz) [inline web/client/plugins/Identify.jsx:237]: Intercepting in the epic you can remove this

File: web/client/components/misc/FeatureInfoTriggerSelector.jsx [hunk 3]
```diff
@@ -15,16 +15,23 @@ const {FormControl, FormGroup, ControlLabel} = require('react-bootstrap');
 class FeatureInfoTriggerSelector extends React.Component {
     static propTypes = {
         trigger: PropTypes.string,
-        onTriggerChange: PropTypes.func
+        onTriggerChange: PropTypes.func,
+        onSetMapTrigger: PropTypes.func
     }
+
+    onChange = (event) => {
+        this.props.onTriggerChange(event);
```
Comment (offtherailz) [inline web/client/components/misc/FeatureInfoTriggerSelector.jsx:23]: Intercepting the event trigger you can remove also this

**Comment by AGMETEOR** [comment]
> Testing to save an existing map with hover set up, I don't see the data persisted in the JSON and in fact when I save, the map reloads and the mouse event is set to click again.
> 
> Please pay attention to the main focus of the issue to solve. You should add this to your unit tests.

I am testing the changes and it seems to work. I set up hovering and then save the map and the hovering is still working.

File: web/client/reducers/map.js [hunk 4]
```diff
@@ -128,6 +130,9 @@ function mapConfig(state = {eventListeners: {}}, action) {
         }
         return data;
     }
+    case SET_MAP_TRIGGER: {
```
Comment (offtherailz) [inline web/client/reducers/map.js:133]: Why you added this? I see it is not read anywhere.

File: web/client/reducers/__tests__/map-test.js [hunk 5]
```diff
@@ -258,4 +259,9 @@ describe('Test the map reducer', () => {
         expect(state.zoom).toEqual(zoom);
         expect(state.mapStateSource).toEqual(null);
     });
+    it('tests SET_MAP_TRIGGER to set trigger in mapInfoConfiguration', () => {
```
Comment (offtherailz) [inline web/client/reducers/__tests__/map-test.js:262]: if state in map is not needed, you can also remove this test

**Comment by offtherailz** [comment]
@AGMETEOR please backport this PR to 2020.02.xx, thank you :+1:

**Pull request merged**